### ISSUE TITLE
[Analyzer] MPI-Checker: doesn't classify some functions, causing false positives

### ISSUE THREAD

**Issue by SuperSamus** [llvm/llvm-project#208107]
The file [MPIFunctionClassifier.cpp](https://github.com/llvm/llvm-project/blob/main/clang/lib/StaticAnalyzer/Checkers/MPI-Checker/MPIFunctionClassifier.cpp) classifies many of MPI's functions, but not all of them.

For instance, `MPI_Iscatterv` is completely missing, and thus:
```c
#include <mpi.h>

int main(int argc, char **argv) {
  MPI_Init(&argc, &argv);

  MPI_Request request = MPI_REQUEST_NULL;
  MPI_Iscatterv(NULL, NULL, NULL, MPI_DATATYPE_NULL, NULL, 0, MPI_DATATYPE_NULL,
                0, MPI_COMM_WORLD, &request);
  MPI_Wait(&request, MPI_STATUS_IGNORE);

  MPI_Finalize();
}
```

The analyzer complains with a false positive:
```
test.c:10:3: warning: Request 'request' has no matching nonblocking call.  [clang-analyzer-optin.mpi.MPI-Checker]
    6 |   MPI_Request request = MPI_REQUEST_NULL;
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    7 |   // Ignore the pointlessness of this.
    8 |   MPI_Iscatterv(NULL, NULL, NULL, MPI_DATATYPE_NULL, NULL, 0, MPI_DATATYPE_NULL,
    9 |                 0, MPI_COMM_WORLD, &request);
   10 |   MPI_Wait(&request, MPI_STATUS_IGNORE);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test.c:10:3: note: Request 'request' has no matching nonblocking call.
    6 |   MPI_Request request = MPI_REQUEST_NULL;
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    7 |   // Ignore the pointlessness of this.
    8 |   MPI_Iscatterv(NULL, NULL, NULL, MPI_DATATYPE_NULL, NULL, 0, MPI_DATATYPE_NULL,
    9 |                 0, MPI_COMM_WORLD, &request);
   10 |   MPI_Wait(&request, MPI_STATUS_IGNORE);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

While there is no false positive with, for instance, `MPI_Iscatter`.

**Comment by steakhal** [comment]
Wow, I'm kind of surprised that this is still a thing. I've been around for 6-7 years and I don't think I have ever seen a bug report (or patch for that matter) about the MPI checker. So this isn't dead after all?
